test/test_eos_mgusup.cpp

```diff
@@ -0,0 +1,527 @@
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+        Kokkos::fence();
+        Kokkos::deep_copy(h_cv, v_cv);
+#endif // PORTABILITY_STRATEGY_KOKKOS
+        THEN("The returned T(rho, e) should be equal to the true value") {
+          array_compare(num, density, temperature, h_cv, cv_true, "Density",
+                        "Temperature");
+        }
+      }
+
+      WHEN("A alpha(rho, T) lookup is performed") {
+        portableFor(
+            "alpha", 0, num, PORTABLE_LAMBDA(const int i) {
+              v_alpha[i] = eos2.TExpansionCoeffFromDensityTemperature(v_density[i],
+                                                                      v_temperature[i]);
+            });
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+        Kokkos::fence();
+        Kokkos::deep_copy(h_alpha, v_alpha);
+#endif // PORTABILITY_STRATEGY_KOKKOS
+        THEN("The returned alpha(rho, T) should be equal to the true value") {
+          array_compare(num, density, temperature, h_alpha, alpha_true, "Density",
+                        "Temperature");
+        }
+      }
+    }
+  }
+}
+SCENARIO("MGUsup EOS SetUp", "[VectorEOS][MGUsupEOS]") {
+  GIVEN("Parameters for a MGUsup EOS") {
+    // Unit conversions
+    constexpr Real Mbcc_per_g = 1e12;
+    // MGUsup parameters for copper
+    Real rho0 = 8.93;
+    Real T0 = 298.0;
+    Real B0 = 1.3448466 * Mbcc_per_g;
+    Real BP0 = 4.956;
+    Real A0 = 5.19245e-05;
+    Real Cv0 = 0.383e-05 * Mbcc_per_g;
+    constexpr Real E0 = 0.0;
+    constexpr Real S0 = 5.05e-04 * Mbcc_per_g;
+    // Create the EOS
+    EOS host_eos = MGUsup(rho0, T0, B0, BP0, A0, Cv0, E0, S0);
+    EOS eos = host_eos.GetOnDevice();
+    eos.Finalize();
+
+    WHEN("Faulty/not set parameter rho0 is given") {
+      rho0 = -1.0;
+      REQUIRE_MAYBE_THROWS(MGUsup(rho0, T0, B0, BP0, A0, Cv0, E0, S0));
+      THEN("An error message should be written out") {}
+    }
+    WHEN("Faulty/not set parameter T0 is given") {
+      T0 = -1.0;
+      REQUIRE_MAYBE_THROWS(MGUsup(rho0, T0, B0, BP0, A0, Cv0, E0, S0));
+      THEN("An error message should be written out") {}
+    }
+    WHEN("Faulty/not set parameter B0 is given") {
+      B0 = -1.0;
+      REQUIRE_MAYBE_THROWS(MGUsup(rho0, T0, B0, BP0, A0, Cv0, E0, S0));
+      THEN("An error message should be written out") {}
+    }
+    WHEN("Faulty/not set parameter BP0 is given") {
+      BP0 = -1.0;
+      REQUIRE_MAYBE_THROWS(MGUsup(rho0, T0, B0, BP0, A0, Cv0, E0, S0));
+      THEN("An error message should be written out") {}
+    }
+    WHEN("Faulty/not set parameter Cv0 is given") {
+      Cv0 = -1.0;
+      REQUIRE_MAYBE_THROWS(MGUsup(rho0, T0, B0, BP0, A0, Cv0, E0, S0));
+      THEN("An error message should be written out") {}
+    }
+    WHEN("Faulty/not set parameter A0 is given") {
+      A0 = -10000000.0;
+      REQUIRE_MAYBE_THROWS(MGUsup(rho0, T0, B0, BP0, A0, Cv0, E0, S0));
+      THEN("An error message should be written out") {}
+    }
```

Comment:
:+1: nice to see this test.

Reply:
Oops, I have the wrong variable names ;). Those are for Vinet. They have the correct values (I did a halfway check but forgot to go all the way) but I will change them to what I use for MGUsup.

Reply:
Fixed